deps/jerry/jerry-core/api/jerry.c

```diff
@@ -3505,6 +3510,83 @@ jerry_get_backtrace_max_depth (void)
   return 10;
 }
 
+bool
+jerry_enable_cpu_profiling (void)
+{
+  jerry_assert_api_available ();
+
+#ifndef JERRY_CPU_PROFILER
+  return false;
+#else
+  return true;
+#endif
+}
+
+/**
+ * start CPU profiling
+ * return false if profiler feature disabled or profiler already started.
+ */
+bool
+jerry_start_cpu_profiling (const char *path,
+                           double duration) /**< millisecond duration, no limit if <= 0 */
+{
+  jerry_assert_api_available ();
+
+#ifndef JERRY_CPU_PROFILER
+  JERRY_UNUSED (path);
+  JERRY_UNUSED (duration);
+  return false;
+#else
+
+  if (JERRY_CONTEXT (cpu_profiling_fp))
+  {
+    return false;
+  }
+
+  FILE *fp = fopen (path, "w");
+  if (fp == NULL)
+  {
+    return false;
+  }
+
+  JERRY_CONTEXT (cpu_profiling_fp) = fp;
+
+  JERRY_CONTEXT (cpu_profiling_start_time) = jerry_port_get_current_time ();
+  JERRY_CONTEXT (cpu_profiling_duration) = duration;
+  return true;
+#endif
+}
+
+bool
+jerry_stop_cpu_profiling (void)
+{
+  jerry_assert_api_available ();
+
+#ifndef JERRY_CPU_PROFILER
+  return false;
+#else
+
+  if (!JERRY_CONTEXT (cpu_profiling_fp))
+  {
+    return false;
+  }
+
+  if (JERRY_CONTEXT (cpu_profiling_fp))
+  {
+    fclose (JERRY_CONTEXT (cpu_profiling_fp));
+    JERRY_CONTEXT (cpu_profiling_fp) = NULL;
+  }
+
+  jerry_flush_parser_dump_file ();
```

Comment:
This function seems to be removed from the latest tree.

Reply:
To read the dump file in clang, use https://github.com/Rokid/ShadowNode/blob/master/deps/jerry/jerry-core/api/jerry.c#L3447, and in JavaScript, see https://github.com/Rokid/ShadowNode/blob/master/src/js/iotjs.js#L53-L60.